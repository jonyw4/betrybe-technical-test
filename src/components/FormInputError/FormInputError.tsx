import * as React from 'react';

/**
 * Text that displays when has an error on the input of the form
 */
export function FormInputError({
  message = 'Esse campo é obrigatório'
}: FormInputErrorProps) {
  return <div className="form-input-error">{message}</div>;
}

interface FormInputErrorProps {
  message?: string;
}
