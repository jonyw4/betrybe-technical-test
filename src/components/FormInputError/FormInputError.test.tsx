import * as React from 'react';
import renderer from 'react-test-renderer';
import { FormInputError } from './FormInputError.component';

describe('components > FormInputError', () => {
  it('should renders a default FormInputError correctly', () => {
    const tree = renderer.create(<FormInputError />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should renders a FormInputError with custom message correctly', () => {
    const tree = renderer
      .create(<FormInputError message="Campo diferenciado" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
