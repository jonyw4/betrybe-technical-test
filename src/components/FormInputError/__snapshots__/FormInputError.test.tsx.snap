// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > FormInputError should renders a FormInputError with custom message correctly 1`] = `
<div
  className="form-input-error"
>
  Campo diferenciado
</div>
`;

exports[`components > FormInputError should renders a default FormInputError correctly 1`] = `
<div
  className="form-input-error"
>
  Esse campo é obrigatório
</div>
`;
